aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -291,6 +292,33 @@ Tensor& sum_out(Tensor& result, const Tensor& self, DimnameList dim,
   return at::sum_out(result, self, dimnames_to_positions(self, dim), keepdim, opt_dtype);
 }
 
+Tensor& nansum_out(Tensor& result, const Tensor& self, IntArrayRef dim,
+                       bool keepdim, optional<ScalarType> opt_dtype) {
+  // For integral types, use existing sum as
```

Comment:
Since this only dispatches on float types (later), what about adding a test here for self being complex and throwing an informative error if it is? Something like "nansum does not support complex inputs"?